fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -303,55 +309,87 @@ public DataApiRequest(
     }
 
     /**
-     * Update sort field string by truncating the dateTime if it is present.
+     * To check whether dateTime column request is first one in the sort list or not.
      *
-     * @param sorts  String of sort columns
+     * @param sortColumns  map of columns and its direction
      *
-     * @return Updated sort column if dateTime exists. Otherwise return as it is received.
+     * @return True if dateTime column is first one in the sort list. False otherwise
      */
-    protected String truncateTimeSort(String sorts) {
+    protected Boolean isDateTimeFirstSortField(Map<String, SortDirection> sortColumns) {
+        if (sortColumns != null) {
+            List<String> columns = new ArrayList(sortColumns.keySet());
```

Comment:
I really don't like this. The contract for the input here is a Map, which is inherently unsorted, so assuming that you can get the keySet and take the first item in the list to get the first item from the sort list is incorrect. I understand that behind the scenes, sortColumns is a LinkedHashMap, which causes this to actually return the desired result, but that isn't captured here in the contract. If you really expect that, then I think you need to change the signature here to REQUIRE a LinkedHashMap. Otherwise, you need to change this to some other Collection that does enforce ordering that isn't a Map.